environments/exercise2.r

```diff
@@ -12,3 +12,17 @@ myGet <- function(x) {
   NULL
 }
 
+recursiveGet <- function(name, env = parent.frame()) {
+  stopifnot(is.character(name), length(name) == 1)
+  if (identical(env, emptyenv())) {
+    # base case
+    warning(paste(name," not found"))
+    NULL
+  } else if (is.element(name,ls(env))) {
+    # success case
+    env[[`name`]]
```

Comment:
In R, backticks create infix functions.  Like ``+`` is addition in R, and lets you do `1 + 1`, instead of ``+`(1, 1)` (which also works).